libs/prelude/Prelude/Functor.idr

```diff
@@ -22,3 +23,5 @@ infixl 4 <$>
 (<$>) : Functor f => (func : a -> b) -> f a -> f b
 func <$> x = map func x
 
+Functor (Pair a) where
+  map f (x, y) = (x, f y)
```

Comment:
@melted Since `Pair` takes two type parameters, we have for a fixed `a` that `Pair a : Type -> Type` as required, so we can implement `Functor` for it.

@justjoheinz It's not so much that the person implementing `Functor` is making the arbitrary choice that it's the second element of the pair that gets mapped over, but rather the person who defines the constructor of `Pair a b`.

The following fails to typecheck:

```idris
implementation [first] Functor (\x => Pair x a) where
  map f (x, y) = (f x, y)
```

which would be the only way I could implement a functor instance causing the first element to get mapped over. So in other words, unless it were possible to write such an instance, someone wanting to implement Functor for a partially-applied Pair has no choice but to make the function `f` apply to the *second* element.

If `x : Pair a b` were such that `fst x : b` (which would be quite unnatural) then the natural implementation of Functor would have the natural result that it's the first element that gets transformed.